### PR TITLE
[DRAFT PR] Kat/image grid component

### PR DESCRIPTION
**Pull request by katm726**
This is a draft PR for the image grid component. I'm not sure what the best way to create asymmetry between the sizes of the images is. I thought it might be a good idea to require that the component get a specific number of photos, because then I could make a rule like "every 3 photo in the array is extra long" or something like that, but I'm hoping there's a better way to do it. Any ideas?

Image grid with numOfRows prop set to 2 :
image

Image grid with numOfRows prop set to 3: 
image

Image grid with numOfRows prop set to 4:
image

### CODE REVIEW

**Review by pablonajera — COMMENTED** [review]
One more thing, when using this in Component Preview, create a wrapper div with an explicit width and height. It'll serve as the container for the grid.

File: components/imageGrid/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import Masonry from 'react-masonry-css';
```
[the remark this answers is not in the document]
Comment (katm726) [reply components/imageGrid/index.js:2]: Ohhh my bad, yes I did use yarn add. I'll push the other two files soon

File: components/imageGrid/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    throw new Error('Image grid component requires numOfRows');
+  }
+  if (photos == null) {
+    throw new Error('Image grid component requires photos array');
+  }
+  return (
+    <Segment basic textAlign="center">
+      <Masonry
+        className={imageGridStyles.fullWidth}
+        breakpointCols={numOfRows}
```
[the remark this answers is not in the document]
Comment (katm726) [reply components/imageGrid/index.js:17]: Can you take a look at the pics I added? I was confused while working with this component bc breakpointCols certainly implies number of columns, but when I change the number it seems to impact the num of rows.  Let me know if there's something I'm not understanding yet here lol the breakpointCols prop is throwing me off

File: components/imageGrid/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import Masonry from 'react-masonry-css';
+import imageGridStyles from '@app/components/imageGrid/imageGrid.module.css';
+import { Segment } from 'semantic-ui-react';
+
+export function ImageGrid({ numOfRows, photos }) {
```
[the remark this answers is not in the document]
Comment (katm726) [reply components/imageGrid/index.js:6]: Why?

File: components/imageGrid/index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  }
+  return (
+    <Segment basic textAlign="center">
+      <Masonry
+        className={imageGridStyles.fullWidth}
+        breakpointCols={numOfRows}
+        columnClassName={imageGridStyles.fullWidth}
+      >
+        {photos.map((photo, index) => (
+          <img src={photo} key={index} height={100} width={100}></img>
```
[the remark this answers is not in the document]
Comment (katm726) [reply components/imageGrid/index.js:21]: :0 omg. 

But how do I set the image width and height to something other than an explicit number? I tried it like this: 

`</img>`

but it won't let me. How would you do it?

**Comment by pablonajera** [comment]
closing in favor of #27

**Pull request closed without merging**